providers/oidc.go

```diff
@@ -114,7 +115,8 @@ func (p *OIDCProvider) createSessionState(ctx context.Context, token *oauth2.Tok
 	}
 
 	if claims.Email == "" {
-		return nil, fmt.Errorf("id_token did not contain an email")
+		// TODO: Try getting email from /userinfo before falling back to Subject
+		claims.Email = claims.Subject
```

Comment:
If `claims.Email == ""` then would `claims.Verified` necessarily be unset? We might want to make a `user := claims.Subject` variable and then overwrite it with `claims.Email` if it is set and then we can check for `claims.Email != ""` before `claims.Verified != nil`, WDYT?

Reply:
The verified takes it value from "email_verified" field, I would assume that if the backend does not provide use with an email, it also will not be saying whether email is verified. As I read the code the intent was to ignore the email if it was unverified. Maybe there are some backends out there that do send email_verified even without email. Don't know.

I'd rather do minimum possible right now and leave that handling to whoever implements getting email with separate request.